f/connectors/globalforestwatch/gfw_alerts.py

```diff
@@ -193,3 +219,129 @@ def format_alerts_as_geojson(alerts: list, type_of_alert: str):
 
     logger.info("GeoJSON formatting complete.")
     return geojson
+
+
+def prepare_gfw_metadata(alerts: list, type_of_alert: str, minimum_date: str):
+    """
+    Prepare GFW alerts metadata for database storage.
+
+    This function creates metadata records for all months from minimum_date
+    to the current month, tracking the alerts found or creating zero-count
+    records for months with no alerts. This ensures we track the full detection
+    range showing when the algorithm has been running.
+
+    Parameters
+    ----------
+    alerts : list
+        A list of alerts fetched from GFW API.
+    type_of_alert : str
+        The type of alert being processed (e.g., 'nasa_viirs_fire_alerts', 'gfw_integrated_alerts').
+    minimum_date : str
+        The minimum date used for the query (format: YYYY-MM-DD).
+
+    Returns
+    -------
+    list of dict
+        A list containing metadata records for all months from minimum_date to current month.
+    """
+    logger.info("Preparing GFW alerts metadata.")
+
+    # Parse the minimum_date to get start year and month
+    date_parts = minimum_date.split("-")
+    start_year = int(date_parts[0])
+    start_month = int(date_parts[1])
+
+    # Get current date for end range
+    current_date = datetime.now()
+    end_year = current_date.year
+    end_month = current_date.month
+
+    # Map alert types to descriptions
+    alert_descriptions = {
+        "nasa_viirs_fire_alerts": "fires",
+        "gfw_integrated_alerts": "deforestation",
+        "glad_alerts": "deforestation",
+        "radd_alerts": "deforestation",
+        "modis_alerts": "deforestation",
+    }
+
+    description_alerts = alert_descriptions.get(type_of_alert, "deforestation")
+
+    # Count alerts by month
+    alerts_by_month = {}
+    for alert in alerts:
+        # Extract date from alert (format varies by alert type)
```

Comment:
I am not seeing where we handle any format other than `YYYY-MM`. Please explain.